src/midi/midi_mt32.cpp

```diff
@@ -346,33 +444,63 @@ static std::deque<std_fs::path> get_selected_dirs()
 	return rom_dirs;
 }
 
-static std::string get_selected_model()
+static std::string get_model_setting()
 {
-	const auto section = static_cast<Section_prop *>(control->GetSection("mt32"));
+	const auto section = static_cast<Section_prop*>(control->GetSection("mt32"));
 	assert(section);
 	return section->Get_string("model");
 }
 
-static std::set<const LASynthModel*> has_models(const MidiHandler_mt32::service_t& service,
-                                                const std_fs::path& dir)
+static std::set<const LASynthModel*> find_models(const MidiHandler_mt32::service_t& service,
+                                                 const std_fs::path& dir)
 {
-	std::set<const LASynthModel *> models = {};
-	for (const auto &model : all_models)
-		if (model->InDir(service, dir))
+	std::set<const LASynthModel*> models = {};
+	for (const auto& model : all_models) {
+		if (model->InDir(service, dir)) {
 			models.insert(model);
+		}
+	}
 	return models;
 }
 
-static std::optional<model_and_dir_t> load_model(
-        const MidiHandler_mt32::service_t& service,
-        const std::string& selected_model, const std::deque<std_fs::path>& rom_dirs)
+static std::optional<ModelAndDir> load_model(const MidiHandler_mt32::service_t& service,
+                                             const std::string& wanted_model_name,
+                                             const std::deque<std_fs::path>& rom_dirs)
 {
-	const bool is_auto = (selected_model == "auto");
-	for (const auto &model : all_models)
-		if (is_auto || model->Matches(selected_model))
-			for (const auto &dir : rom_dirs)
-				if (model->Load(service, dir))
+	auto [load_first_available, candidate_models] =
+	        [&]() -> std::pair<bool, const std::vector<const LASynthModel*>&> {
+		// Symbolic model names resolve the first available model in the
+		// given category.
```

Comment:
this is pretty confusing; it's an anonymous lambda function that populates a bool and a reference to a vector of models, and then we use those in the search loop below.

I don't have suggestions to simplifying this closer to the prior code version, but it would be handy to document it just briefly because the initial block is an eyeful! :slightly_smiling_face: 

```c++
	auto [load_first_available, candidate_models] =
	        [&]() -> std::pair<bool, const std::vector<const LASynthModel*>&>
```

Reply:
Yeah, lack of proper type inference of the return value... 😩 I can simplify to get rid of the lambda and use two mutable vars for the flag & list of candidate models if that helps.

I mean if you find things confusing at the mechanical code level. At the conceptual level, I find the logic clear with the "candidate models". But maybe I could break this into two functions with some code duplication to get rid of the boolean altogether: one function for loading versioned models, and another one for looking up the "best" symbolic model. Probably that's for the best.

Reply:
The mechanical code (inside the lands) and the candidate loop all look good (and very similar to the original code); but I just had to paused at the lamda and think about what it's doing / why it's needed. 

So yeah, no need to change the code, but just a one-liner comment to say what we're trying to get or setup with the lamda. 

(This, being an anonymous Landa, means there isn't a function name that can act as self-documenting code.. so the reader only see the variable names and return type).

Just a minor comment though 👍